### PR TITLE
Archive TAP files

### PR DESCRIPTION
**Pull request by JeromeJu**
Archive TAPfiles zip

Related: https://github.com/adoptium/aqa-tests/issues/3286

Grinder link: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/3757/

Sign off by: [jeromeqju@gmail.com](mailto:jeromeqju@gmail.com)

### CODE REVIEW

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -911,6 +912,17 @@ def archiveAQACertFile() {
 	}
 }
 
+def archiveTapFiles() {
+	def TAPzipFile = "aqa-tests/TKG/TAPFiles.zip";
+	if (!params.ARTIFACTORY_SERVER) {
+		echo "ARTIFACTORY_SERVER is not set. Saving TAPzipFile on jenkins."
+		zip zipFile: TAPzipFile, archive: true, glob: "**/*.tap"
```
[the remark this answers is not in the document]
Comment (JeromeJu) [reply buildenv/jenkins/JenkinsfileBase]: I've rebased according to [#3325](https://github.com/adoptium/aqa-tests/pull/3333), but not sure how's the expected way to push the tap files to the TAPzipFile artifact with the generic `archiveFile` function.

File: buildenv/jenkins/JenkinsfileBase [hunk 2]
```diff
@@ -696,11 +696,15 @@ def post(output_name) {
...
+		  def TAPzipFile = "aqa-tests/TKG/TAPFiles.zip"
 
             //call the archive function for each file
             archiveFile(testenvFile)
-	    archiveFile(shaFile)
+            archiveFile(shaFile)
             archiveFile(AQACertFile)
+		  // TODO: how to push the tap files to TAPzipFile with the same archiveFile function?
+		  // archiveArtifacts artifacts: "**/*.tap", fingerprint: true, allowEmptyArchive: true
+		  archiveFile(TAPzipFile)
```
Comment (llxia) [inline buildenv/jenkins/JenkinsfileBase:707]: Please remove line 695 - 699 and update line 702 - 707 to directly pass in the file as the parameter.

**Comment by llxia** [comment]
Convert this PR to draft as it is not ready yet.

**Comment by JeromeJu** [comment]
@llxia thank you for your reminder.
And my apologies for the delay and blocking on the issue. I forgot this PR as in draft and mistook some commit that I have made but not yet pushed should suffice. Will be prioritizing and putting more efforts on the certification related issues.

**Comment by llxia** [comment]
The above Grinder is gone. Reran https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/746/ and the TAP is archived successfully
```
17:20:11  [consumer_1] Deploying artifact: https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Grinder/746/Grinder.tap
```

**Review by llxia — APPROVED** [review]
LGTM

**Comment by smlambert** [comment]
Running a named job from this branch to see what the tap file name looks like when archived: https://ci.adoptopenjdk.net/view/Test_functional/job/Test_openjdk11_hs_sanity.functional_x86-64_linux/228 - looks good!

**Review by smlambert — APPROVED** [review]

**Pull request merged**